src/app/utils.test.ts

```diff
@@ -0,0 +1,119 @@
+import {
+    createUploadDirIfNotExists,
+    deleteFile,
+    generateAuthorizationHeaders,
+    isValidUUID,
+    saveFile,
+    UPLOAD_DIR,
+} from './utils'
+import mockFs from 'mock-fs'
+import path from 'path'
+import jwt from 'jsonwebtoken'
+import { describe, expect, it, vi } from 'vitest'
+import fs from 'fs'
+
+describe('Utils', () => {
+    describe('createUploadDirIfNotExists', () => {
+        it('should create the upload directory if it does not exist', async () => {
+            // Ensure the directory does not exist
+            mockFs({})
+            expect(() => fs.statSync(UPLOAD_DIR)).toThrow()
+
+            createUploadDirIfNotExists()
+
+            // Verify the directory was created
+            expect(fs.statSync(UPLOAD_DIR).isDirectory()).toBe(true)
+        })
+
+        it('should not create the upload directory if it already exists', async () => {
+            // Ensure the directory already exists
+            mockFs({
+                [UPLOAD_DIR]: {},
+            })
+
+            // Try to create it again
+            createUploadDirIfNotExists()
+
+            // Verify the directory is still intact
+            const files = fs.readdirSync(UPLOAD_DIR)
+            expect(files).toEqual([])
+        })
+    })
```

Comment:
This test is kind of odd to me in that I'm not sure what changes to `utils.ts` could ever cause it to fail since we are using `{ recursive: true }` (e.g. in a TDD flow, would it have ever started in a failed state that would necessitate a code change?). Is it adding any value / can it be removed?

Reply:
great point - i thought this was needed to get 100% coverage in the file but i was wrong! I am a fan of having only whats necessary in tests 👍